stack-graphs/src/stitching.rs

```diff
@@ -938,27 +974,96 @@ impl ForwardPartialPathStitcher {
     /// [`process_next_phase`][] manually.
     ///
     /// [`process_next_phase`]: #method.process_next_phase
-    pub fn find_all_complete_partial_paths<I>(
+    pub fn find_all_complete_partial_paths<I, F>(
         graph: &StackGraph,
         partials: &mut PartialPaths,
         db: &mut Database,
         starting_nodes: I,
         cancellation_flag: &dyn CancellationFlag,
-    ) -> Result<Vec<PartialPath>, CancellationError>
+        mut visit: F,
+    ) -> Result<(), CancellationError>
     where
         I: IntoIterator<Item = Handle<Node>>,
+        F: FnMut(&StackGraph, &mut PartialPaths, &PartialPath),
     {
-        let mut result = Vec::new();
         let mut stitcher =
             ForwardPartialPathStitcher::from_nodes(graph, partials, db, starting_nodes);
         while !stitcher.is_complete() {
             cancellation_flag.check("finding complete partial paths")?;
             let complete_partial_paths = stitcher
                 .previous_phase_partial_paths()
                 .filter(|partial_path| partial_path.is_complete(graph));
-            result.extend(complete_partial_paths.cloned());
+            for path in complete_partial_paths {
+                visit(graph, partials, path);
+            }
```

Comment:
You might consider passing the entire vector into a single call to `visit`, and letting the visitor function take care of iterating as needed.  That will be a bit more efficient, since you won't have the function call overhead for each path that's found.  And I don't think it will be that much more tedious for the caller.  (And then if the caller is for instance extending into a different vec it can use `extend`, which can do a bulk copy)

Reply:
I tried that at first, but got into trouble with borrows. In the end the code on the client side could not simply call `extend` and became more verbose then doing it like this.